apps/functions/src/functions/enroll.ts

```diff
@@ -114,7 +113,9 @@ export const enroll = Functions.endpoint
 
         const classes = await classesRepository.getMany(selectedClasses);
 
-        const classGroups = await semester.groups.getMany(selectedClassGroups);
+        const classGroups = await SemesterRepository.of(
+            SUMMER_2023_SEMESTER
+        ).groups.getByClassIds(selectedClasses);
```

Comment:
During enrollment, the selected class groups are now derived from the selected classes.